tcp.go

```diff
@@ -22,6 +22,28 @@ var DefaultConnectTimeout = 5 * time.Second
 
 var log = logging.Logger("tcp-tpt")
 
+const keepAlivePeriod = 30 * time.Second
+
+type canKeepAlive interface {
+	SetKeepAlive(bool) error
+	SetKeepAlivePeriod(time.Duration) error
+}
+
+var _ canKeepAlive = &net.TCPConn{}
+
+func tryKeepAlive(conn net.Conn, keepAlive bool) {
+	keepAliveConn, ok := conn.(canKeepAlive)
+	if !ok {
+		log.Errorf("Can't set TCP keepalives.")
+	}
+	if err := keepAliveConn.SetKeepAlive(keepAlive); err != nil {
+		log.Errorf("Failed to enable TCP keepalive: %s", err)
```

Comment:
It's good to start off with an error, but this _may_ have false positives (e.g., connection closed?). However, I _assume_ that this method won't fail until we actually call `close` ourselves.

Also, return?